freeipa/src/main/java/com/sequenceiq/freeipa/converter/instance/template/InstanceTemplateRequestToTemplateConverter.java

```diff
@@ -63,6 +65,15 @@ public Template convert(InstanceTemplateRequest source, CloudPlatform cloudPlatf
                             }
                         }
                 );
+        Optional.ofNullable(source.getAzure())
+                .map(AzureInstanceTemplateParameters::getEncryptionParameters)
+                .ifPresent(encryptionParameters -> {
+                            if (Objects.nonNull(encryptionParameters.getDiskEncryptionSetId())) {
+                                attributes.put(AzureInstanceTemplate.DISK_ENCRYPTION_SET_ID, encryptionParameters.getDiskEncryptionSetId());
+                                attributes.put(AzureInstanceTemplate.MANAGED_DISK_ENCRYPTION_WITH_CUSTOM_KEY_ENABLED, EncryptionType.CUSTOM);
+                            }
+                        }
+                );
```

Comment:
This certainly works. The question is if it really makes sense to pass all these encryption details in the FreeIPA req (i.e. `AzureInstanceTemplateParameters`). For DH creation, I pulled the encryption details directly from the `EnvironmentResponse`, without explicitly making changes to the DH API; see https://github.com/hortonworks/cloudbreak/pull/10371/files#diff-d1e272ac5092f0d0d5fb69f48ea4275188e174e69d864a901d6ff44c5f9f7968 .